runtime/sema/check_composite_declaration.go

```diff
@@ -819,6 +843,14 @@ func (checker *Checker) declareCompositeLikeMembersAndValue(
 			nestedCompositeDeclarationVariable :=
 				checker.valueActivations.Find(identifier.Identifier)
 
+			if identifier.Identifier == ast.ResourceDestructionDefaultEventName {
```

Comment:
`identifier.Identifier == ast.ResourceDestructionDefaultEventName` is repeated in a few places, maybe create a function for it, like e.g. `IsResourceDestructionDefaultEventIdentifier`